client-spark/spark-3/src/main/scala/org/apache/spark/shuffle/celeborn/ShuffleFetchFailureReportTaskCleanListener.scala

```diff
@@ -19,10 +19,25 @@ package org.apache.spark.shuffle.celeborn
 
 import org.apache.spark.scheduler.{SparkListener, SparkListenerStageCompleted}
 
-class ShuffleFetchFailureReportTaskCleanListener extends SparkListener {
+import org.apache.celeborn.common.internal.Logging
+
+class ShuffleFetchFailureReportTaskCleanListener extends SparkListener with Logging {
   override def onStageCompleted(stageCompleted: SparkListenerStageCompleted): Unit = {
-    SparkUtils.removeStageReportedShuffleFetchFailureTaskIds(
-      stageCompleted.stageInfo.stageId,
-      stageCompleted.stageInfo.attemptNumber())
+    val stageId = stageCompleted.stageInfo.stageId
+    val stageAttemptId = stageCompleted.stageInfo.attemptNumber()
+    val shuffleFetchFailureTaskIds =
+      SparkUtils.removeStageReportedShuffleFetchFailureTaskIds(stageId, stageAttemptId)
+    if (shuffleFetchFailureTaskIds != null) {
+      val taskSetManager = SparkUtils.getTaskSetManager(stageId, stageAttemptId)
```

Comment:
TSM should be used within the scheduler imp lock

Reply:
thanks